src/dependencies/InstallerSequence.ts

```diff
@@ -13,16 +13,28 @@ export class InstallerSequence {
 		}, []);
 	}
 
-	public async install(location: Location, shouldUpdate: boolean, progressListener: ProgressListener): Promise<Location> {
+	public async install(location: Location, shouldUpdate: boolean, progressListener: ProgressListener, confirm:() => Promise<void>): Promise<Location> {
 		let index = 0;
+		let askedForConfirmation = false;
 		const total = this.installers.length;
 		for (const installer of this.installers) {
-			location = await installer.install(location, shouldUpdate, (fraction, message) => {
+			function intermediateListener(fraction: number, message: string) {
 				progressListener(
 					(index + fraction) / total,
 					`${message} (step ${index + 1} of ${total})`
 				);
-			});
+			}
+			function intermediateConfirm(): Promise<void> {
+				// only ask once
+				if (askedForConfirmation) {
+					return Promise.resolve();
+				} else {
+					askedForConfirmation = true;
+					return confirm();
+				}
+			}
+
+			location = await installer.install(location, shouldUpdate, intermediateListener, intermediateConfirm);
```

Comment:
It might be easier to do `await confirm();` at line 18 instead of adding `askedForConfirmation` and `intermediateConfirm`.

Reply:
The idea is that every installer can decide on its own whether confirmation is necessary or not. For example, the downloader only asks for permissions if actually something is downloaded (as it does not exists yet)

Reply:
I see, nice!